Makefile

```diff
@@ -1,195 +1,10 @@
-# Minimum version numbers for software required to build IPFS
-IPFS_MIN_GO_VERSION = 1.7
-IPFS_MIN_GX_VERSION = 0.6
-IPFS_MIN_GX_GO_VERSION = 1.1
+# General tools
 
-GOTAGS =
-GOTAGS += "" # we have to have always at least one tag, empty tag works well
+SHELL=PATH=$(PATH) /bin/sh
 
-GOFLAGS =
-GOTFLAGS =
+PROTOC = protoc --gogo_out=. --proto_path=.:/usr/local/opt/protobuf/include:$(dir $@) $<
 
-export IPFS_REUSEPORT=false
-export GOFLAGS
-export GOTFLAGS
+# enable second expansion
+.SECONDEXPANSION:
```

Comment:
We require GNU make either way even right now, I don't know if it is on purpose or not.

For BSD based systems we recommend using `gmake`.

I could do without this flag, I think, not 100% sure right now.